benchmark/README.md

```diff
@@ -0,0 +1,90 @@
+# Benchmark
+
+## Algorithm Benchmarks
+
+In order to perform algorithm benchmarks, `make` to build the executable __benchmark__ and then use
+it as follows:
+
+```
+Usage: benchmark [-vaofsnpyz]                         
+   -v    verbose                                   
+   -a    algorithm name                            
+   -o    objective function name                   
+   -f    output timing file name                   
+   -s    output solution file name                 
+   -n    number of iterations to run               
+   -p    population size                  
+   -y    minimum values                            
+   -z    maximum values                            
+                                               
+   Example:                                        
+     ./benchmark -a "hgwosca" -o "sum" -n 3 -d 10 -p 30 -y -120 -z 100 -s "solution.txt" -f "timings.txt"
+```
+\-f and \-o parameters take absolute/relative paths as well.
+
+This usage information is also shown using `bench -h`.
+
+## Algorithm Benchmarks
+This executable runs our algorithms and measures the cycles performed for them  
+to complete for a given parameter set. All algorithms and objective functions which are registered in 
+__benchmark.cpp__ are runnable. Parameters are handed in via command line.
+
+### Output
+\-f and \-o parameters specify the output files for timing and solution logging respectively.
+
+The file created by \-f is formatted like
+```
+iteration, cycles
+0, 320299
+1, 239529
+2, 172374
+... until (n - 1)  
+```
+The file created by \-o is formatted like 
+```
+dimension, value
+0, -120
+1, 98
+... until (d - 1)  
+```
+
+## Operation Benchmarks
+
+__TODO: CURRENTLY THIS SECTION IS NOT TRUE, SORRY JAKOB :)__
```

Comment:
Yes maybe we should keep it separated. Otherwise we end up with a massive exec that is super polyvalent but impossible to use. There are already a shit ton of args, no need to make it even more.